docs/source/posts/2024/2024-08-06-week6-wachiou-bouraima.rst

```diff
@@ -0,0 +1,92 @@
+WEEK 6: Code reviews, relining and crush challenges
+===================================================
+
+.. post:: August 6, 2024
+   :author: Wachiou BOURAIMA
+   :tags: google
+   :category: gsoc
+
+Hello everyone,
+As my Google Summer of Code (GSoC) 2024 journey progresses, week6 has brought me a series of technical challenges and accomplishments. My main focus has been on code reviews, rebasing and commits squasing, with a few notable lessons learned along the way.
+
+
+Code Reviews and Merging ``Pull Requests``
+-------------------------------------------
+
+One of the main activities this week was receiving and addressing feedback on several of my pull requests. Notably, my mentor `Serge Koudoro <https://github.com/skoudoro/>`_, reviewed and merged the PRs related to the ``warn_on_args_to_kwargs`` decorator and the application of the ``warn_on_args_to_kwargs`` decorator across various modules. The merging of these PRs was a critical step in ensuring that our codebase adhered to the project's evolving standards for clarity and maintainability.
+
+
+Rebasing and Squashing: Overcoming Challenges
+---------------------------------------------
+
+- I performed rebasing and squashing to integrate the latest changes and consolidate commits. This process was challenging due to several conflicts that arose. Resolving these conflicts required a deep dive into Git’s functionality, including:
+
+  - **Conflict Resolution:** Manually resolving merge conflicts that affected several files.
+
+  - **Understanding Git Operations:** Gained hands-on experience with rebasing and squashing, which improved my grasp of version control workflows.
+
+  - **Commit Consolidation:** multiple commits into a single commit to streamline the commit history and enhance readability.
+
+.. code-block:: bash
```

Comment:
Perhaps you could mention why you introduce this codeblock

Reply:
By the way, I just wanted to review the command I had to use. I think maybe I should remove it. What do you think ?